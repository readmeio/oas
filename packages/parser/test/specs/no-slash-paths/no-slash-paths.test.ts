import { describe, expect, it } from 'vitest';

import { validate } from '../../../src/index.js';

describe('no-slash-paths', () => {
  it('should provide clear error message for paths missing leading slashes', async () => {
    await expect(validate(require.resolve('./invalid-no-slash-paths.json'))).resolves.toMatchSnapshot();
  });
});
