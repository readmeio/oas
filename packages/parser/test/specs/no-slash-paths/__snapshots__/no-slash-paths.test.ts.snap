// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-slash-paths > should provide clear error message for paths missing leading slashes 1`] = `
{
  "additionalErrors": 0,
  "errors": [
    {
      "message": "Entries in the OpenAPI \`paths\` object must begin with a leading slash.",
    },
  ],
  "specification": "OpenAPI",
  "valid": false,
  "warnings": [],
}
`;
